app/collins/solr/CollinsQueryLanguage.scala

```diff
@@ -27,7 +27,7 @@ case class CQLQuery(select: SolrDocType, where: SolrExpression) {
  * itself into a Solr Query
  */
 sealed trait SolrQueryComponent {
-  protected[solr] def traverseQueryString(): String = traverseQueryString(true)
```

Comment:
Just a quick skim not really a through check, `toplevel` param is unused.

Also, caching has plenty of tests and this seems like one that is very do-able, ymmv